File: 2000 98 & ME/QuickStart/howto/samples/Services/MessageQueue/mqctrl/vb/mqctrl.vb
Imports System
Imports System.Messaging

Public Class MQCtrl
    Public Shared Sub  Main()
        Dim appName As String = Environment.GetCommandLineArgs()(0)

        if(Environment.GetCommandLineArgs().Length <> 4) then
            Console.WriteLine("Usage: ")
            Console.WriteLine("    " + appName +" <queue> <command> <value>")
            Console.WriteLine("")
            Console.WriteLine("        <Command> =")
            Console.WriteLine("            l: change label")
            Console.WriteLine("            s: change queue size")
            Console.WriteLine("            j: change journal size")
            Exit Sub
        End if
        
        Dim mqPath As String = ".\" + Environment.GetCommandLineArgs()(1)
        if(Not MessageQueue.Exists(mqPath)) then
            Console.WriteLine("The queue " + mqPath + " does not exist")
            Exit Sub
        End if
        
        Dim mq As MessageQueue = new MessageQueue(mqPath)

    Dim command As String = Environment.GetCommandLineArgs()(2)
    Dim value As String = Environment.GetCommandLineArgs()(3)
        if(command = "l") then
            mq.Label = value
        elseif(command = "s") then
            mq.MaximumQueueSize = Int32.FromString(value)
        else if(command = "j") then
            mq.MaximumJournalSize = Int32.FromString(value)
        end if
    end sub
end class

File: 2000 98 & ME/QuickStart/howto/samples/Services/MessageQueue/mqctrl/mqctrl.cs
using System;
using System.Messaging;

public class MQCtrl {
    public static void Main(String[] args) 
    {
        string appName = Environment.GetCommandLineArgs()[0];

        if ( args.Length != 3 ) {
            Console.WriteLine("Usage: ");
            Console.WriteLine("\t" + appName +" <queue> <command> <value>");
            Console.WriteLine("");
            Console.WriteLine("\t\t<command> =");
            Console.WriteLine("\t\t\tl: change label");
            Console.WriteLine("\t\t\ts: change queue size");
            Console.WriteLine("\t\t\tj: change journal size");
            return;
        }

        string mqPath = ".\\" + args[0];
        if ( !MessageQueue.Exists(mqPath) ) {
            Console.WriteLine("The queue " + mqPath + " does not exist");
            return;     
        }

        MessageQueue mq = new MessageQueue(mqPath);

        string command = args[1];

        if ( command == "l" ) {
            mq.Label = args[2]; 
        } else if ( command == "s" ) {
            mq.MaximumQueueSize = UInt32.FromString(args[2]);
        } else if ( command == "j" ) {
            mq.MaximumJournalSize = UInt32.FromString(args[2]);
        }
    }
}